stan/math/prim/mat/fun/mdivide_right_tri.hpp

```diff
@@ -38,10 +38,11 @@ namespace stan {
       } else if (TriView == Eigen::Upper) {
         return transpose(mdivide_left_tri<Eigen::Lower>(transpose(A),
                                                         transpose(b)));
-      } else {
-        throw std::domain_error("triangular view must be Eigen::Lower or "
-                                "Eigen::Upper");
       }
+
+      domain_error("mdivide_left_tri",
+                   "triangular view must be Eigen::Lower or Eigen::Upper",
+                   "", "");
```

Comment:
Do we go with multiple returns and fall-through for errors in general? It's always a confusing style to me but I guess it's common (?)